### PR TITLE
Allow google-auth < 3 and google-api-core < 3

### PR DESCRIPTION
**Pull request by busunkim96**
`google-auth` recently published a 2.0.0 release which removed support for Python 2.7. `google-auth` now requires Python >=3.6. No other breaking changes were made. You can see the full list of changes [here](https://github.com/googleapis/google-auth-library-python/commit/560cf1ed02a900436c5d9e0a0fb3f94b5fd98c55).

I am opening PRs to expand google-auth version ranges for packages that meet either of the following criteria:
* Package has >10,000 monthly downloads as of June 2021
* Package is owned by a Google team

`google-auth` is a dependency of many different libraries that interact with Google APIs. Increasing the time and number of packages with compatible pins on `google-auth` lowers the chance end developers who use multiple libraries will see dependency conflicts. 

If possible, please do not require `google-auth>=2.0.0` or `google-api-core>=2.0.0` until https://github.com/googleapis/google-cloud-python/issues/10566 is resolved, as that will further reduce the likelihood of diamond dependency conflicts.

Googlers, see [this doc](https://docs.google.com/document/d/1euAvUsia_4zf98lNvpwA3K0o2b4y5Xr9xAkyzCnIMzQ/edit) for more information.

Given the number of google-cloud-* packages this library pins directly I also bumped the pin on `google-api-core` to < 3.

**Constraints files**: I looked at https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pinned-constraint-files but am not quite sure what that means for this change.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -319,7 +319,7 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'pandas-gbq<0.15.0',
 ]
 grpc = [
-    'google-auth>=1.0.0, <2.0.0dev',
+    'google-auth>=1.0.0, <3.0.0dev',
```
Comment (uranusjr) [inline setup.py]: (Not directly related to this PR) @potiuk Why was the `dev` suffix added in the first place and can we remove it? This seems to be the only place we are specifying prereleases explicitly (in cdb3f25456e49d0199cd7ccd680626dac01c9be6).

Comment (potiuk) [reply setup.py]: Good point. I have no idea :). I added it > year ago, and without suitable comment (my bad). But I am quite sure, whatever it was then it is not applicable any more and we can change it to <3.0.0 - providing that it passes all the tests.

Comment (uranusjr) [reply setup.py]: @busunkim96 Could you try removing `dev` from the version constraints?

Comment (busunkim96) [reply setup.py]: Done.

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**